crates/ethereum/consensus/src/lib.rs

```diff
@@ -137,8 +136,45 @@ where
     H: BlockHeader,
 {
     fn validate_header(&self, header: &SealedHeader<H>) -> Result<(), ConsensusError> {
-        validate_header_gas(header.header())?;
-        validate_header_base_fee(header.header(), &self.chain_spec)?;
+        let header = header.header();
+        let is_post_merge = self.chain_spec.is_paris_active_at_block(header.number());
+
+        if is_post_merge {
+            if !header.difficulty().is_zero() {
+                return Err(ConsensusError::TheMergeDifficultyIsNotZero);
+            }
+
+            if !header.nonce().is_some_and(|nonce| nonce.is_zero()) {
+                return Err(ConsensusError::TheMergeNonceIsNotZero);
+            }
+
+            if header.ommers_hash() != EMPTY_OMMER_ROOT_HASH {
+                return Err(ConsensusError::TheMergeOmmerRootIsNotEmpty);
+            }
+
+            validate_header_extra_data(header)?;
+        } else {
+            #[cfg(feature = "std")]
+            {
+                let present_timestamp = std::time::SystemTime::now()
+                    .duration_since(std::time::SystemTime::UNIX_EPOCH)
+                    .unwrap()
+                    .as_secs();
+
+                if header.timestamp() >
+                    present_timestamp + alloy_eips::merge::ALLOWED_FUTURE_BLOCK_TIME_SECONDS
+                {
+                    return Err(ConsensusError::TimestampIsInFuture {
+                        timestamp: header.timestamp(),
+                        present_timestamp,
+                    });
+                }
+            }
+
+            validate_header_extra_data(header)?;
```

Comment:
I think we can even move `validate_header_extra_data` out of the if else check? because this is called in both branches